olive/systems/system_config.py

```diff
@@ -30,11 +39,18 @@ class DockerTargetUserConfig(TargetUserConfig):
 
 class AzureMLTargetUserConfig(TargetUserConfig):
     azureml_client_config: AzureMLClientConfig = None
-    aml_compute: str
+    aml_compute: str = None
+    aml_compute_config: AmlComputeConfig = None
     aml_docker_config: AzureMLDockerConfig
     instance_count: int = 1
     is_dev: bool = False
 
+    @validator("aml_compute_config", pre=True)
+    def _validate_aml_compute_config(cls, v, values):
+        if v is None and values.get("aml_compute") is None:
```

Comment:
should not create default AmlComputeConfig, users need provide the name, size, location explicitly